rcl_yaml_param_parser/include/rcl_yaml_param_parser/types.h

```diff
@@ -95,6 +95,33 @@ typedef struct rcl_node_params_s
   size_t capacity_params;  ///< Capacity of parameters in the node
 } rcl_node_params_t;
 
+/// \typedef rcl_param_descriptor_t
+/// \brief param_descriptor_t stores the descriptor of a parameter
+typedef struct rcl_param_descriptor_s
+{
+  char * name;
+  bool * read_only;
+  uint8_t * type;
+  char * description;
+  char * additional_constraints;
+  double * min_value_double;
+  double * max_value_double;
+  double * step_double;
+  int64_t * min_value_int;
+  int64_t * max_value_int;
+  int64_t * step_int;
+} rcl_param_descriptor_t;
+
+/// \typedef rcl_node_params_descriptors_t
+/// \brief node_params_descriptors_t stores all the parameter descriptors of a single node
+typedef struct rcl_node_params_descriptors_s
+{
+  char ** parameter_names;  ///< Array of parameter names (keys)
+  rcl_param_descriptor_t * parameter_descriptors;  ///< Array of corresponding parameter descriptors
+  size_t num_params;  ///< Number of parameters in the node
+  size_t capacity_descriptors;  ///< Capacity of parameters in the node
+} rcl_node_params_descriptors_t;
```

Comment:
@bpwilcox wouldn't adding an `rcl_param_descriptor_t ** parameter_descriptors;` field to the `rcl_node_params_s` struct simplify the code a bit? Using NULL to indicate when a descriptor is not available, precluding parameter descriptors without an associated parameter value.

Reply:
I was initially going to put the together as you recommend, however I do not think we would want the parameter descriptors to require association with a parameter value. In my view, it should be perfectly acceptable to have a yaml file with descriptors only and declared parameters in code may comply with these descriptions.